src/commands/findRoom.ts

```diff
@@ -0,0 +1,361 @@
+	busySince: string;
+	busyUntil: string;
+	isInUse: boolean;
+}
+
+interface GetRoomsResponse {
+	Rooms: Array<[string, string]> | [];
+}
+
+const timeChoices = [
+	{ name: '8:00 AM', value: '08:00:00' },
+	{ name: '8:30 AM', value: '08:30:00' },
+	{ name: '9:00 AM', value: '09:00:00' },
+	{ name: '9:30 AM', value: '09:30:00' },
+	{ name: '10:00 AM', value: '10:00:00' },
+	{ name: '10:30 AM', value: '10:30:00' },
+	{ name: '11:00 AM', value: '11:00:00' },
+	{ name: '11:30 AM', value: '11:30:00' },
+	{ name: '12:00 PM', value: '12:00:00' },
+	{ name: '12:30 PM', value: '12:30:00' },
+	{ name: '1:00 PM', value: '13:00:00' },
+	{ name: '1:30 PM', value: '13:30:00' },
+	{ name: '2:00 PM', value: '14:00:00' },
+	{ name: '2:30 PM', value: '14:30:00' },
+	{ name: '3:00 PM', value: '15:00:00' },
+	{ name: '3:30 PM', value: '15:30:00' },
+	{ name: '4:00 PM', value: '16:00:00' },
+	{ name: '4:30 PM', value: '16:30:00' },
+	{ name: '5:00 PM', value: '17:00:00' },
+	{ name: '5:30 PM', value: '17:30:00' },
+	{ name: '6:00 PM', value: '18:00:00' },
+	{ name: '6:30 PM', value: '18:30:00' },
+	{ name: '7:00 PM', value: '19:00:00' },
+	{ name: '7:30 PM', value: '19:30:00' },
+];
+
+const bldgChoices = [
+	{ name: 'ANY', value: 'ANY' },
+	{ name: 'BNSN', value: 'BNSN' },
+	{ name: 'BRMB', value: 'BRMB' },
+	{ name: 'CTB', value: 'CTB' },
+	{ name: 'EB', value: 'EB' },
+	{ name: 'ELLB', value: 'ELLB' },
+	{ name: 'ESC', value: 'ESC' },
+	{ name: 'HBLL', value: 'HBLL' },
+	{ name: 'JFSB', value: 'JFSB' },
+	{ name: 'JKB', value: 'JKB' },
+	{ name: 'JRCB', value: 'JRCB' },
+	{ name: 'JSB', value: 'JSB' },
+	{ name: 'KMBL', value: 'KMBL' },
+	{ name: 'LSB', value: 'LSB' },
+	{ name: 'MARB', value: 'MARB' },
+	{ name: 'MB', value: 'MB' },
+	{ name: 'MCDB', value: 'MCDB' },
+	{ name: 'MCKB', value: 'MCKB' },
+	{ name: 'MSRB', value: 'MSRB' },
+	{ name: 'SLC', value: 'SLC' },
+	{ name: 'TLRB', value: 'TLRB' },
+	{ name: 'TMCB', value: 'TMCB' },
+	{ name: 'TNRB', value: 'TNRB' },
+	{ name: 'WSC', value: 'WSC' },
+	{ name: 'WVB', value: 'WVB' },
+];
+
+export function convertTo12Hour(time: string): string {
+	const [hours, minutes, seconds] = time.split(':').map(Number);
+	if (hours !== undefined && minutes !== undefined && seconds !== undefined) {
+		const suffix = hours >= 12 ? 'PM' : 'AM';
+		const convertedHours = hours % 12 || 12;
+		return `${convertedHours}:${String(minutes).padStart(2, '0')} ${suffix}`;
+	}
+	return 'ERR';
+}
+
+async function _getRoomsNow(search_type: string, building: string): Promise<GetRoomsResponse> {
+	try {
+		const { data, status } = await axios.get<GetRoomsResponse>(
+			`https://pi.zyancey.com/${search_type}/${building}`
+		);
+		if (status !== 200) throw new Error(`${status}`);
+		return data;
+	} catch (error_) {
+		logger.error('Error in getting Room Info:');
+		logger.error(error_);
+		const error: GetRoomsResponse = {
+			Rooms: [],
+		};
+		return error;
+	}
+}
+
+async function _getRoomsAt(
+	search_type: string,
```

Comment:
It seems to me like `search_type` should probably just be a const inside this function (ditto for the other `getRooms` functions)